subiquity/cloudinit.py

```diff
@@ -100,8 +110,8 @@ def read_legacy_status(stream):
     return None
 
 
-async def get_schema_failure_keys() -> list[str]:
-    """Retrieve the keys causing schema failure."""
+async def get_unknown_keys() -> list[str]:
+    """Retrieve top-level keys causing schema failures, if any."""
 
     cmd: list[str] = ["cloud-init", "schema", "--system"]
     status_coro: Awaitable = arun_command(cmd, clean_locale=True)
```

Comment:
should this be using system_env as well?

Reply:
Yes it should. It won't really matter once we unstage cloud-init but it really should use the external version here regardless.